Upload logs

pakrym-oai opened the pull request:
# External (non-OpenAI) Pull Request Requirements

Before opening this Pull Request, please read the dedicated "Contributing" markdown file or your PR may be closed:
https://github.com/openai/codex/blob/main/docs/contributing.md

If your PR conforms to our contribution guidelines, replace this text with a detailed and high quality description of your changes.

pakrym-oai committed:
Upload logs

lcian commented on codex-rs/feedback/src/lib.rs:
```diff
@@ -238,12 +189,9 @@ impl CodexLogSnapshot {
         envelope.add_item(EnvelopeItem::Event(event));
```
We need to attach a trace ID to the logs and, as explained below, ideally we would attach the same trace ID to the event and each of the logs.

You can either generate a trace ID and add it manually (for the event it needs to go in `event.contexts.trace.trace_id` , as a `TraceContext`, you can see how to add this in `sentry-rust`) or you can use some functions that do this for you.
I would recommend the second approach, as that will also add some default additional contexts/attributes to the event/log.

To use the built-in functions, you would go something like this:
```rust
    sentry::configure_scope(|scope| {
        scope.apply_to_log(&mut log);
        event = scope.apply_to_event(event).unwrap();
    });
```

lcian reviewed (commented):
Hey, Lorenzo from the Sentry SDKs team here. Left some comments.

pakrym-oai committed:
feedback: add trace context to logs and events uploaded to Sentry

Adds trace context (trace_id, span_id) to Sentry events and log items when uploading feedback. Ensures that both main event and logs are correlated via the same trace_id. Logs are now uploaded in a separate envelope when include_logs is true. Removes unnecessary import.
